gnd/src/main/java/com/google/android/gnd/persistence/geojson/GeoJsonParser.java

```diff
@@ -96,8 +96,7 @@ private String readJsonFile(File file) {
 
     for (int i = 0; i < arr.length(); i++) {
       try {
-        JSONObject o = arr.getJSONObject(i);
-        result[i] = o;
+        result[i] = arr.getJSONObject(i);
```

Comment:
When an exception is thrown here `result[i]` will remain uninitialized (`null`). In this case you'd need to `filter` them out in the caller `intersectingTiles()`. A cleaner/simpler approach would be to add them to an `ArrayList` instead of a primitive array.

Reply:
Gotcha, switched to ArrayList